src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -0,0 +1,130 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import * as React from 'react';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+import { Modal as OfficeFabricModal, IModalProps } from 'office-ui-fabric-react/lib/Modal';
+import { FixedGridRow, FixedGridColumn } from '../FixedGrid';
+import Heading from '../Heading';
+import Clickable from '../Clickable';
+import Block, { GutterSize } from '../Block';
+import CloseIcon from '../Icon/icons/Cancel';
+import { IconSize } from '../Icon';
+import ScreenreaderText from '../ScreenreaderText';
+import { join } from '../../util/classNames';
+import { getClassNames } from './ModalDialog.styles';
+import { ModalDialogSize } from './types';
+import { KeyCodes } from '@uifabric/utilities';
+
+export interface ModalDialogProps extends BaseComponentProps {
+  /**
+   * The title text to display at the top of the dialog.
+   */
+  title: string;
+
+  /**
+   * Replace the title with a hidden ScreenreaderText title.
+   * @default false
+   */
+  hideTitle?: boolean;
+
+  /**
+   * Determines whether the dialog is displayed or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * A callback function for when the Dialog is dismissed from the close button or light dismiss.
+   */
+  onDismissed: IModalProps['onDismissed'];
+
+  /**
+   * The close button's aria label.
+   */
+  closeAriaLabel: string;
+
+  /**
+   * The size of the ModalDialog
+   */
+  size: ModalDialogSize;
+}
+
+export interface ModalDialogState {
+  isOpen: boolean;
```

Comment:
Why have `isOpen` as both a prop and a state key? Shouldn't we decide whether this is a controlled or uncontrolled component? I think all other stateful components in YamUI are controlled (e.g. `Dropdown`).

Reply:
Yeah I realized that after discussing with Arturo. I removed state completely and it worked fine but now I'm going to remove everything that I've added as it's no longer necessary.